platform-sdk/docs/proposals/TSS-Library/TSS-Library.md

```diff
@@ -0,0 +1,812 @@
+
+###### `PairingPublicKey`
+**Link**: [PairingPublicKey.java](signature-lib%2FPairingPublicKey.java)
+
+**Description**: A public key generated using the pairings API
+###### `PairingSignature`
+**Link**: [PairingSignature.java](signature-lib%2FPairingSignature.java)
+
+**Description**: A signature generated with the private key that can be verified with the public key
+
+###### `GroupAssignment`
+**Link**: [GroupAssignment.java](signature-lib%2FGroupAssignment.java)
+
+**Description**: An enum to clarify which group public keys and signatures are in, for a given SignatureSchema
+
+
+##### Implementation Note
+The serialization of the elements in this module adds a byte to represent the combination of Curve type and group assignment.
+
+##### Examples
+###### Generating a SignatureSchema
+```java
+SignatureSchema signatureSchema = SignatureSchema.from(Curve.ALT_BN128, GroupAssignament.G1_PUBLIC_KEYS);
+```
+###### Generating a Private Key
+```java
+   PairingPrivateKey pk = PairingPrivateKey.create(signatureSchema, new SecureRandom());
+   
+```
+###### Generating a Public Key
+```java
+   PairingPublicKey pbk = PairingPublicKey.create(pk);
+   
+```
+###### Generating a Signature
+```java
+    byte[] message = new byte[]{};
+    PairingSignature s = PairingSignature.sign(pk, message );
+   
+```
+###### Verifying a Signature
+```java
+static{
+    s.verify(pbk, message);
+}
+```
+##### Constraints
+This module will not depend on hedera-services artifacts, so it cannot include logging, metrics, configuration, or any other helper module from that repo.
+##### Dependencies
+swirlds-cryptography-pairings-API and runtime implementation
+##### Other considerations
+Some unknowns are worth investigating in a follow-up task:
+
+#### Swirlds Crypto TSS Library
+##### Overview
+This library implements the Groth21 TSS-specific primitives.
+
+##### Constraints
+Once built, this library will depend on artifacts constructed in other repositories.
+This library would accept Integer.MAX_VALUE -1 participants.
+
+##### Public API:
+###### `TssService`
+**Description**: This class handles all tss specific operations.
+
+**Link**: [TssService.java](tss%2FTssService.java)
+
+###### `TssParticipantDirectory`
+**Description**: This class holds all information about the participants in the scheme. Including: participants' EC public keys, public shares, private shares, number of shares. 
+
+**Link**: [TssParticipantDirectory.java](tss%2FTssParticipantDirectory.java)
+
+###### `TssMessage`
+**Description**: This class is used in the exchange of secret information between participants. Contains an encrypted message, a polynomial commitment, and a cryptographic proof that can be used to validate this message.
+
+**Link**:[TssMessage.java](tss%2FTssMessage.java)
+
+###### `TssShareId`
+**Description**: This class represents the unique identification of a share in the system. It a FieldElement, so it can be used as input in the polynomial.
```

Comment:
Might be missing a few words around `It a FieldElement` and `the polynomial`.
It's not clear to me what was actually intended for that sentence currently.

Reply:
Half worked through a previous suggestion. should be:
`It contains a FieldElement, so it can be used as input in the polynomial.`